src/components/Inputs/TextFieldWithCustomLabel/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import styled from "styled-components";
+
+import { InputLabel } from "@material-ui/core";
+import { TextField } from "@gnosis.pm/safe-react-components";
+import { pxOrCustomCssUnits } from "utils/cssUtils";
+
+// To match parent's width https://github.com/gnosis/safe-react-components/blob/development/src/inputs/TextField/index.tsx#L24
+const WIDTH = 400;
+
+const Wrapper = styled.div<{ width?: string | number }>`
+  width: ${({ width }) => pxOrCustomCssUnits(width || WIDTH)};
+
+  // Adjust input padding which is normally pushed down to accommodate default label 
+  .MuiFilledInput-input {
+    padding 11px 12px 9px;
```

Comment:
Not happy with this. Doesn't feel right.
Also, doesn't align nicely when adding `endAdornment`.

How can I do this better, the proper way?

Reply:
Seems like the way to go: https://material-ui.com/components/text-fields/#CustomizedInputs.tsx (Bootstrap input)